Spelling

jsoref opened the pull request:
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).


Typos can result in tests failing to test what they intend to test. Visible typos show a lack of polish. Typos can result in poor translations. And typos can mean that when searching for an identifier, things will be missed.

The misspellings have been reported at https://github.com/jsoref/owncloud-android/commit/b8400fb97d0f111356e4954dab2d4c5f0ca141f6#commitcomment-56629989
The action reports that the changes in this PR would make it happy: 
https://github.com/jsoref/owncloud-android/commit/8f731cd790abdfe89eb4b874094d0ac9546d4e3f

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/android?pullRequest=3388) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/android?pullRequest=3388) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/android?pullRequest=3388) it.</sub>

jsoref commented on THIRD_PARTY.txt:
```diff
@@ -11,7 +11,7 @@
 
 This program is free software: you can redistribute it and/or modify
-it under the terms of the GNU General Public License verions 2,
+it under the terms of the GNU General Public License version 2,
```
License files can be incredibly picky about changes, but even so, I'd hope this is a reasonable change.

jsoref commented on owncloudApp/src/main/java/com/owncloud/android/db/UploadResult.java:
```diff
@@ -29,7 +29,7 @@
     CONFLICT_ERROR(4),
     FILE_ERROR(5),
-    PRIVILEDGES_ERROR(6),
+    PRIVILEGES_ERROR(6),
```
api break?

abelgardep replied:
It should not break anything. If you rename it everywhere, it will be fine. We store the value in the database, not the name. 👍

jsoref commented on owncloudApp/src/main/java/com/owncloud/android/presentation/manager/AvatarManager.kt:
```diff
@@ -62,7 +62,7 @@ class AvatarManager : KoinComponent {
             return BitmapUtils.bitmapToCircularBitmapDrawable(appContext.resources, it)
         }
 
-        // Avatar not found in disk caché, fetch from server.
```
maybe there's an interesting story here?

jsoref commented on owncloudApp/src/main/java/com/owncloud/android/utils/PermissionUtil.java:
```diff
@@ -45,7 +45,7 @@ public static boolean shouldShowRequestPermissionRationale(Activity activity, St
      * @param activity The target activity.
      */
-    public static void requestWriteExternalStoreagePermission(Activity activity) {
+    public static void requestWriteExternalStoragePermission(Activity activity) {
```
api break

abelgardep replied:
It should not break anything if you renamed it everywhere as you did 👍

jsoref commented on owncloudApp/src/main/java/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.kt:
```diff
@@ -129,11 +129,11 @@ class ErrorMessageAdapter {
                     if (operation is CreateFolderOperation) f.forbidden(R.string.forbidden_permissions_create)
                     if (operation is MoveFileOperation) f.forbidden(R.string.forbidden_permissions_move)
                     if (operation is CopyFileOperation) f.forbidden(R.string.forbidden_permissions_copy) else f.format(
-                        R.string.filename_forbidden_charaters_from_server
```
this probably breaks localizations. I generally don't touch localizations as projects typically have special rules for updates.

abelgardep replied:
Yes, it will unless we rename it on every single strings file.

So, we have two options:
1. Rename it on every single strings file on this PR
2. Revert this change and rename it on another PR

Choose the one you are more comfortable with.

abelgardep replied:
Changes applied, renamed on every strings file 👍

jsoref commented on owncloudData/src/test/java/com/owncloud/android/data/oauth/datasource/RemoteOAuthDataSourceTest.kt:
```diff
@@ -97,10 +97,10 @@ class RemoteOAuthDataSourceTest {
         } returns tokenResponseResult
 
-        val tokenResponsed = remoteOAuthDataSource.performTokenRequest(OC_TOKEN_REQUEST_ACCESS)
+        val tokenResponded = remoteOAuthDataSource.performTokenRequest(OC_TOKEN_REQUEST_ACCESS)
```
This should probably be `tokenResponse` (corrections were automatically picked by Google Sheets)

abelgardep replied:
Yes, that's it. Could you rename it to `tokenResponse`?

abelgardep commented:
Thanks for the contribution @jsoref, we will take a look 👍

abelgardep commented:
Hey @jsoref, could you check the comments? Thanks!

jsoref commented:
@abelgardep: I believe I've addressed the comments. I'm not really available until October, but you should be able to push to my branch if you need to, so please feel free to do so if necessary.

abelgardep reviewed (approved):
LGTM

jesmrec commented:
Checked locally the UI tests. They mainly pass, using same emulator as CI i noticed that the test `loginBasic_callLoginBasic_trimUsername` fails. But, also in the `master` branch. We will have to put an eye there.

For me, no problem to move this forward.
